src/test/java/org/jboss/logmanager/handlers/PeriodicRotatingFileHandlerTests.java

```
@@ -0,0 +1,146 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+
+public class PeriodicRotatingFileHandlerTests extends AbstractHandlerTest {
+    private final static String FILENAME = "periodic-rotating-file-handler.log";
+
+    private final String logFile = BASE_LOG_DIR.toString() + "/" + FILENAME;
+
+    private final SimpleDateFormat rotateFormatter = new SimpleDateFormat(".yyyy-MM-dd");
+    private PeriodicRotatingFileHandler handler;
+
+    @Before
+    public void createHandler() throws FileNotFoundException {
+        // Create the handler
+        File logFile = new File(this.logFile);
+        handler = new PeriodicRotatingFileHandler(logFile, rotateFormatter.toPattern(), false);
+        handler.setFormatter(FORMATTER);
+    }
+
+    @After
+    public void closeHandler() {
+        handler.close();
+        handler = null;
+    }
+
+    @Test
+    public void testRotate() throws Exception {
+        final Calendar cal = Calendar.getInstance();
+        final String rotatedPath = BASE_LOG_DIR.toString() + "/" + FILENAME + rotateFormatter.format(cal.getTime());
+        File rotatedFile = new File(rotatedPath);
+        testRotate(cal, rotatedFile);
+    }
+
+    @Test
+    public void testOverwriteRotate() throws Exception {
+        final Calendar cal = Calendar.getInstance();
+        final String rotatedPath = BASE_LOG_DIR.toString() + "/" + FILENAME + rotateFormatter.format(cal.getTime());
+        
+        // Create the rotated file to ensure at some point it gets overwritten
+        File rotatedFile = new File(rotatedPath);
+        
+        if (rotatedFile.exists())
+            rotatedFile.delete();
+        
+        try {
+            FileWriter fwriter = new FileWriter(rotatedFile.getAbsolutePath());
+	    BufferedWriter  writer = new BufferedWriter(fwriter);
+            writer.write("Adding data to the file");
+            writer.close(); 
+        } catch (Exception e) {
+            e.printStackTrace();
```

This should throw the exception so the test will fail.